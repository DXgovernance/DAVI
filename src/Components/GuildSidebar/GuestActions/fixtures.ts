import { GuestActionsProps } from './GuestActions';

export const withData: GuestActionsProps = {
  userWalletAddress: '0x0000000000000000000000000000000000000000',
  onShowStakeModal: jest.fn(),
  onShowWalletModal: jest.fn(),
};
