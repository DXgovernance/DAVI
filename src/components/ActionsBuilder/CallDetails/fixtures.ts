import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
