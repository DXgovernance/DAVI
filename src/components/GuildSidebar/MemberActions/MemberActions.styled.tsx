import styled from 'styled-components';
import { IconButton } from 'components/Primitives/Button';

export const UserActionButton = styled(IconButton)`
  border-radius: 50px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  font-weight: 600;
  & > div:first-child {
    display: flex;
  }
`;

export const IconHolder = styled.span`
  display: flex;
  justify-content: center;

  @media only screen and (min-width: 768px) {
    margin-right: 0.3rem;
  }

  img {
    border-radius: 50%;
    margin-right: 0;
  }
`;

export const VotingPower = styled.div`
  background-color: #282b30;
  color: #fff;
  border-radius: 32px;
  padding: 2px 8px;
  font-weight: 500;
  font-size: 14px;
`;
