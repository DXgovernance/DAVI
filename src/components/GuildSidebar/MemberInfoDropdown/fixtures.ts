import { BigNumber } from 'ethers';
import moment from 'moment';
import { MemberInfoDropdownProps } from './MemberInfoDropdown';

export const closedWithoutData: Omit<MemberInfoDropdownProps, 'onClose'> = {
  isOpen: false,
};

export const openWithoutData: Omit<MemberInfoDropdownProps, 'onClose'> = {
  isOpen: true,
};

export const closedWithData: Omit<MemberInfoDropdownProps, 'onClose'> = {
  isOpen: false,
  guildToken: {
    decimals: 18,
    name: 'Guild Token',
    symbol: 'GT',
    totalSupply: BigNumber.from('1000000000'),
  },
  isRepGuild: false,
  unlockedAt: moment().add(1, 'day'),
  userVotingPower: BigNumber.from('250000000'),
  userVotingPowerPercent: 50,
};

export const openWithData: Omit<MemberInfoDropdownProps, 'onClose'> = {
  ...closedWithData,
  isOpen: true,
};
