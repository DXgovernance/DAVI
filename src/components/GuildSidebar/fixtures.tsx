import { GuildSidebarProps } from './GuildSidebar';

const TestDiv = () => <div>Test</div>;

export const withData: GuildSidebarProps = {
  guildName: 'Test Guild',
  numberOfMembers: 10,
  actions: TestDiv(),
};
