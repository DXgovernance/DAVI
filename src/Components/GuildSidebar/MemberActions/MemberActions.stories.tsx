import { ComponentStory, ComponentMeta } from '@storybook/react';
import { propsWithData } from './fixtures';
import { MemberActions } from './MemberActions';

export default {
  title: 'Sidebars/GuildSidebar/MemberActions',
  component: MemberActions,
} as ComponentMeta<typeof MemberActions>;

const Template: ComponentStory<typeof MemberActions> = args => (
  <MemberActions {...args} />
);

export const Simple = Template.bind({});
Simple.args = propsWithData;

export const Loading = Template.bind({});
Loading.args = {};
