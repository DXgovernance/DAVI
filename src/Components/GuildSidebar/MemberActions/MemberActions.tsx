import { ERC20Info } from 'hooks/Guilds/ether-swr/erc20/useERC20Info';
import { shortenAddress } from 'utils';
import Avatar from 'Components/Avatar';
import { Loading } from '../../Primitives/Loading';
import { Moment } from 'moment';
import { useState } from 'react';
import {
  IconHolder,
  UserActionButton,
  VotingPower,
} from './MemberActions.styled';
import { BigNumber } from 'ethers';
import { ENSAvatar } from 'Components/Types';
import { MemberInfoDropdown } from '../MemberInfoDropdown';

export interface MemberActionsProps {
  guildToken?: ERC20Info;
  isRepGuild?: boolean;
  userWalletAddress?: string;
  userVotingPower?: BigNumber;
  userEnsAvatar?: ENSAvatar;
  userVotingPowerPercent?: number;
  unlockedAt?: Moment;
  onWithdraw?: () => void;
  onShowStakeModal?: () => void;
}

export const MemberActions: React.FC<MemberActionsProps> = ({
  isRepGuild,
  guildToken,
  userWalletAddress,
  userEnsAvatar,
  userVotingPower,
  userVotingPowerPercent,
  unlockedAt,
  onWithdraw,
  onShowStakeModal,
}) => {
  const [isInfoDropdownOpen, setIsInfoDropdownOpen] = useState(false);

  return (
    <MemberInfoDropdown
      isOpen={isInfoDropdownOpen}
      onClose={() => setIsInfoDropdownOpen(false)}
      guildToken={guildToken}
      isRepGuild={isRepGuild}
      userVotingPower={userVotingPower}
      userVotingPowerPercent={userVotingPowerPercent}
      unlockedAt={unlockedAt}
      onWithdraw={onWithdraw}
      onShowStakeModal={onShowStakeModal}
    >
      <UserActionButton
        iconLeft
        onClick={() => setIsInfoDropdownOpen(!isInfoDropdownOpen)}
      >
        <div>
          <IconHolder>
            <Avatar
              src={userEnsAvatar?.imageUrl}
              defaultSeed={userWalletAddress}
              size={18}
            />
          </IconHolder>
          <span>
            {userWalletAddress ? (
              userEnsAvatar?.ensName || shortenAddress(userWalletAddress)
            ) : (
              <Loading loading text />
            )}
          </span>
        </div>
        <VotingPower>
          {userVotingPowerPercent != null ? (
            `${userVotingPowerPercent}%`
          ) : (
            <Loading loading text skeletonProps={{ width: '40px' }} />
          )}
        </VotingPower>
      </UserActionButton>
    </MemberInfoDropdown>
  );
};
