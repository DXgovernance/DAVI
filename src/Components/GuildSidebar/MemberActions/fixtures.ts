import { BigNumber } from 'ethers';
import moment from 'moment';
import { MemberActionsProps } from './MemberActions';

export const propsWithData: MemberActionsProps = {
  guildToken: {
    decimals: 18,
    name: 'Guild Token',
    symbol: 'GT',
    totalSupply: BigNumber.from('1000000000'),
  },
  userWalletAddress: '0x1234567890123456789012345678901234567890',
  userEnsAvatar: {
    imageUrl: '/assets/favicon-32x32.png',
    ensName: 'test.eth',
  },
  isRepGuild: false,
  unlockedAt: moment().add(1, 'day'),
  userVotingPower: BigNumber.from('250000000'),
  userVotingPowerPercent: 50,
};
