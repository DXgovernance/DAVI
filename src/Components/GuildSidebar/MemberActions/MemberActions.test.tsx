import { MemberActions } from './MemberActions';
import { render } from 'utils/tests';
import { propsWithData } from './fixtures';

describe('MemberActions', () => {
  it('Should match snapshot without data', () => {
    const { container } = render(<MemberActions />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot with data', () => {
    const { container } = render(<MemberActions {...propsWithData} />);
    expect(container).toMatchSnapshot();
  });
});
