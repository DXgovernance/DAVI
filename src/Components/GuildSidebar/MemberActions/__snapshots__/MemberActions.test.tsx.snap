// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberActions Should match snapshot with data 1`] = `
<div>
   
  <div
    class="sc-pGacB sc-kstqJO kELkBy fkHEBq"
    data-testid="member-actions-button"
  >
    <button
      class="sc-dlfnuX sc-hKgJUU sc-gKseQn dEJhZx eRtASc eHKpXr"
    >
      <div>
        <span
          class="sc-iBPTik lkybgw"
        >
          <img
            alt="Avatar"
            class="sc-eCstlR sc-jSgvzq ebJAYl fbWVQg"
            data-testid="avatar"
            size="18"
            src="/assets/favicon-32x32.png"
          />
        </span>
        <span>
          test.eth
        </span>
      </div>
      <div
        class="sc-fubCzh kFYzWV"
      >
        50%
      </div>
    </button>
    <div
      class="sc-pGacB sc-bqyKOL kELkBy eA-dQQq"
    >
      
      <div
        class="sc-hBEYId gPHLnk"
      >
        <div
          class="sc-fodVek jyvpfu"
        >
          votingPower
           
          <span>
            50%
          </span>
        </div>
        <div
          class="sc-fodVek jyvpfu"
        >
          locked
           
          <span>
            0 GT
          </span>
        </div>
        <div
          class="sc-fodVek jyvpfu"
        >
          unlockedIn
           
          <span>
            a day
          </span>
        </div>
        <button
          class="sc-dlfnuX sc-fFucqa dEJhZx hyGtHU"
        >
          increaseVotingPower
        </button>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`MemberActions Should match snapshot without data 1`] = `
<div>
   
  <div
    class="sc-pGacB sc-kstqJO kELkBy fkHEBq"
    data-testid="member-actions-button"
  >
    <button
      class="sc-dlfnuX sc-hKgJUU sc-gKseQn dEJhZx eRtASc eHKpXr"
    >
      <div>
        <span
          class="sc-iBPTik lkybgw"
        >
          <div
            class="sc-bdfBQB"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="width: 18px; height: 18px; border-radius: 50%; --base-color: #333; --highlight-color: #555;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </span>
        <span>
          <div
            class="sc-bdfBQB"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="width: 100px; height: 16px; border-radius: 50px; --base-color: #333; --highlight-color: #555;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </span>
      </div>
      <div
        class="sc-fubCzh kFYzWV"
      >
        <div
          class="sc-bdfBQB"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="width: 40px; --base-color: #333; --highlight-color: #555;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
    </button>
    <div
      class="sc-pGacB sc-bqyKOL kELkBy eA-dQQq"
    >
      
      <div
        class="sc-hBEYId gPHLnk"
      >
        <div
          class="sc-fodVek jyvpfu"
        >
          votingPower
           
          <span>
            <div
              class="sc-bdfBQB"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="width: 40px; --base-color: #333; --highlight-color: #555;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </span>
        </div>
        <div
          class="sc-fodVek jyvpfu"
        >
          locked
           
          <span>
            <div
              class="sc-bdfBQB"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="width: 40px; --base-color: #333; --highlight-color: #555;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </span>
        </div>
        <div
          class="sc-fodVek jyvpfu"
        >
          unlockedIn
           
          <span>
            <div
              class="sc-bdfBQB"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="width: 40px; --base-color: #333; --highlight-color: #555;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </span>
        </div>
        <button
          class="sc-dlfnuX sc-fFucqa dEJhZx hyGtHU"
        >
          increaseVotingPower
        </button>
      </div>
    </div>
  </div>
   
</div>
`;
